farmdata2_modules/fd2_tabs/resources/FarmOSAPI.spec.js

```diff
@@ -1,74 +1,215 @@
+            cy.intercept("GET","/log?type=farm_seeding&page=1").as('second')
+            cy.intercept("GET","/log?type=farm_seeding&page=2").as('third')
+        
+            let firstCalls=0
+            let secondCalls=0
+            let thirdCalls=0
 
-        cy.wait(1)
-        .then(() => {
-            cy.wait('@first').should(() => {
-                cy.wrap(testArray).should('have.length.gt',0)
-                firstCalls++
-            })
+            cy.wait(1)
+            .then(() => {
+                cy.wait('@first').should(() => {
+                    cy.wrap(testArray).should('have.length.gt',0)
+                    firstCalls++
+                })
 
-            cy.wait('@second').should(() => {
-                // Just by getting here we know the second page was requested.
+                cy.wait('@second').should(() => {
+                    // Just by getting here we know the second page was requested.
 
-                // Check that data made it into testArray.
-                // Note: depending on timing this may not run until after any
-                // subsequent calls.
-                cy.wrap(testArray).should('have.length.gt',100)
-                secondCalls++
-            })
+                    // Check that data made it into testArray.
+                    // Note: depending on timing this may not run until after any
+                    // subsequent calls.
+                    cy.wrap(testArray).should('have.length.gt',100)
+                    secondCalls++
+                })
 
-            cy.wait('@third').should(() => {
-                cy.wrap(testArray).should('have.length.gt',200)
-                thirdCalls++
-            })
+                cy.wait('@third').should(() => {
+                    cy.wrap(testArray).should('have.length.gt',200)
+                    thirdCalls++
+                })
 
-            getAllPages("/log?type=farm_seeding", testArray)
+                getAllPages("/log?type=farm_seeding", testArray)
 
+            })
+            .then(() => {
+                expect(firstCalls).to.equal(1)
+                expect(secondCalls).to.equal(1)
+                expect(thirdCalls).to.equal(1)
+            })
+        })
+    })
+    
+    context('test maping id to name functions', () => {
+        it('getIDToUserMap function returns a map', () => {
+            cy.wrap(getIDToUserMap()).as('map')
+            cy.get('@map').should((idToNameMap) => {
+                expect(idToNameMap).to.not.be.null
+                expect(idToNameMap).to.be.a('Map')
+            })
+        })
+        it('getIDToUserMap returns correct keys', () => {
+            cy.wrap(getIDToUserMap()).as('map')
+            cy.get('@map').should((idToNameMap) => {
+                expect(idToNameMap).to.have.all.keys(null, '6','8','7', '4', '9', '5', '1', '3', '155')
+            })
+        })
+        it('getIDToUserMap correct maps key to value', () => {
```

Comment:
I would suggest removing Anonymous from the tests as that is not an important user in the system.  It would be better to test agains 2 or 3 of the actual users (manager1 or 2, worker1 ... worker4, guest, restws1.